File: yht0827/1월/1월 8일/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
	public static long solution(int n, int k) {
		long answer = 0;
		StringBuilder num = new StringBuilder();
		List<Long> list = new ArrayList<>();

		while (n > 0) {
			num.insert(0, (n % k));
			n /= k;
		}

		String[] split = num.toString().split("0");

		for (String s : split) {
			if (!(s.length() == 0)) {
				System.out.println(s);
				list.add(Long.parseLong(s));
			}
		}

		for (Long l : list) {
			if (checkPrime(l)) {
				answer++;
			}
		}

		return answer;
	}

	public static boolean checkPrime(long num) {
		if (num == 1) {
			return false;
		} else if (num == 2) {
			return true;
		}

		int limit = (int)Math.sqrt(num);
		for (int i = 2; i <= limit; ++i) {
			if (num % i == 0) {
				return false;
			}
		}
		return true;
	}
}

File: zuzubibi/old/1월/1월 8일/k 진수에서 소수 개수 구하기.py
import math
def is_prime_number(x):
    if x <2:
        return False
    for i in range(2, int(math.sqrt(x))+1):
        if x % i ==0:
            return False
    return True

def solution(n, k):
    result = 0
    jinsu = ''
    while n > 0:
        n, mod = divmod(n,k)
        jinsu += str(mod)
    jinsu = jinsu[::-1]

    jinsu = jinsu.split('0')

    for i in jinsu:
        if i and is_prime_number(int(i)):
            result += 1

    return result



print(solution(437674,3)) # 3
print(solution(110011, 10)) # 2